PWGHF/D2H/Macros/ML/train_d2h.py

```diff
@@ -0,0 +1,544 @@
+
+        # output
+        self.outdir = config["output"]["dir"]
+        self.extension = config["plots"]["extension"]
+        self.column_to_save_list = config["output"]["column_to_save_list"]
+        self.log_file = config["output"]["log_file"]
+
+    def __check_input_consistency(self):
+        """
+        Helper method to check self consistency of inputs
+        """
+
+        # class balance
+        if self.share not in ("equal", "all_signal"):
+            print(f"\033[91mERROR: class_balance option {self.share} not implemented\033[0m")
+            sys.exit()
+        if self.share == "all_signal" and len(self.bkg_factor) != len(self.pt_bins):
+            print("\033[91mERROR: bkg_factor must be defined for each pT bin!\033[0m")
+            sys.exit()
+        # training
+        if self.training_vars is None:
+            print("\033[91mERROR: training columns must be defined!\033[0m")
+            sys.exit()
+        # hyper-parameters options
+        if not isinstance(self.hyper_pars, list):
+            print("\033[91mERROR: hyper-parameters must be defined or be a list containing an empty dict!\033[0m")
+            sys.exit()
+        if not isinstance(self.hyper_pars[0], dict):
+            print("\033[91mERROR: hyper-parameters must be a list of dict!\033[0m")
+            sys.exit()
+        if len(self.hyper_pars) != len(self.pt_bins):
+            print("\033[91mERROR: hyper-parameters definition does not match pT binning!\033[0m")
+            sys.exit()
+        if not isinstance(self.hyper_pars_opt["hyper_par_ranges"], dict):
+            print("\033[91mERROR: hyper_pars_opt_config must be defined!\033[0m")
+            sys.exit()
+        if not self.binary:
+            if not (self.roc_auc_average in ["macro", "weighted"] and self.roc_auc_approach in ["ovo", "ovr"]):
+                print("\033[91mERROR: selected ROC configuration is not valid!\033[0m")
+                sys.exit()
+
+            if self.roc_auc_average == "weighted":
+                self.score_metric += f"_{self.roc_auc_approach}_{self.roc_auc_average}"
+            else:
+                self.score_metric += f"_{self.roc_auc_approach}"
+
+    def __fill_list_input_files(self):
+        """
+        Helper method to fill a dictionary with lists of input files for each class
+        """
+
+        file_lists = {}
+        for cand_type in self.indirs:
+            file_lists[cand_type] = []
+            if self.indirs[cand_type] is None:
+                continue
+            for indir in self.indirs[cand_type]:
+                file = os.path.join(indir, f"{cand_type}_{self.channel}.{self.infile_extension}")
+                if os.path.isfile(file):
+                    file_lists[cand_type].append(file)
+                else:
+                    print(
+                        "\033[91mERROR: missing file, did you prepare the samples?\n"
+                        "If not, do python3 prepare_samples.py CONFIG\033[0m"
```

Comment:
This conflicts with the current macro for preparing the samples, which is `PrepareSamples.C`.

Reply:
This is solved with the new (python) version of the preparatio script